lib/gulp-image-converter.js

```diff
@@ -0,0 +1,64 @@
+var path = require('path');
+var _ = require('underscore');
+var through = require('through2');
+var lwip = require('lwip');
+var gutil = require('gulp-util');
+
+function convert(options, file, cb) {
+  var format = path.extname(file.path).slice(1);
+  var targetFormat = options.format || format;
+
+  lwip.open(file.contents, format, function(err, image) {
+    if (err) {
+      return cb(err);
+    }
+
+    image.toBuffer(
+      targetFormat,
+      options.formatParams || {},
+      function(err, buffer) {
+        if (err) {
+          return cb(err);
+        }
+
+        // Note that vinyl 0.5 supports file.extname for manipulations
+        // like this, but we're not yet using 0.5 at the time
+        // of writing this, so we'll need to replace the extension
+        // ourselves.
+        file.path = require('replace-ext')(file.path, '.' + targetFormat);
```

Comment:
Cool thanks Atul!  Whichever works better for you I guess.  It's just I tend to miss this kind of inline todo notes myself so yea... as long as you remember :grin: